src/pip/_internal/utils/misc.py

```diff
@@ -764,16 +765,52 @@ def unpack_file(
         )
 
 
+def make_command(*args):
+    # type: (Union[str, HiddenText, CommandArgs]) -> CommandArgs
+    """
+    Create a CommandArgs object.
+    """
+    command_args = []  # type: CommandArgs
+    for arg in args:
+        # Check for list instead of CommandArgs since CommandArgs is
+        # only known during type-checking.
+        if isinstance(arg, list):
+            command_args.extend(arg)
+        else:
+            # Otherwise, arg is str or HiddenText.
+            command_args.append(arg)
+
+    return command_args
+
+
 def format_command_args(args):
-    # type: (List[str]) -> str
+    # type: (Union[List[str], CommandArgs]) -> str
     """
     Format command arguments for display.
     """
-    return ' '.join(shlex_quote(arg) for arg in args)
+    # For HiddenText arguments, display the redacted form by calling str().
+    # Also, we don't apply str() to arguments that aren't HiddenText since
+    # this can trigger a UnicodeDecodeError in Python 2 if the argument
+    # has type unicode and includes a non-ascii character.  (The type
+    # checker doesn't ensure the annotations are correct in all cases.)
```

Comment:
Reg the type-checker: I think we're supposed to use `mypy.Text` for things that would be unicode on Python 2.

Reply:
This is _supposed_ to always be `str` (if everything is working correctly), but because our type-checking isn't 100%, it's possible something of `unicode` type can slip through the cracks (e.g. because of `# type: ignore` comments at some of our annotations). One example is the `_get_win_folder_with_ctypes()` function in `appdirs.py` that is annotated with `str` but returns `unicode` in Python 2. In cases that slip through the cracks, the point was not to force a crash if we don't need to..